tests/unit/lib/queuePopulator/kafkaLogConsumer/ListRecordStream.js

```diff
@@ -0,0 +1,133 @@
+            {
+                opType: 'update',
+                expected: 'put'
+            },
+            {
+                opType: 'replace',
+                expected: 'put'
+            },
+            {
+                opType: 'delete',
+                expected: 'delete'
+            },
+            {
+                opType: 'unsupported',
+                expected: undefined
+            }].forEach(params => {
+                const { opType, expected } = params;
+                it(`Should return correct operation type (${opType})`, done => {
+                    const type = listRecordStream._getType(opType);
+                    assert.strictEqual(type, expected);
+                    return done();
+                });
+            });
+    });
+
+    describe('_transform', () => {
+        it('Should correct format entry', done => {
+            listRecordStream.write(kafkaMessage);
+            listRecordStream.once('data', data => {
+                assert.deepEqual(data, {
+                    timestamp: new Date(kafkaMessage.timestamp),
+                    db: 'example-bucket',
+                    entries: [{
+                        key: 'example-key',
+                        type: 'put',
+                        value: {
+                            field: 'value'
+                        },
+                    }],
+                });
+                return done();
+            });
+        });
+        it('Should not fail if format is invalid', done => {
+            listRecordStream.write(InvalidKafkaMessage);
+            listRecordStream.once('data', data => {
+                assert.deepEqual(data, {
+                    timestamp: new Date(InvalidKafkaMessage.timestamp),
+                    db: undefined,
+                    entries: [{
+                        key: undefined,
+                        type: undefined,
+                        value: undefined,
+                    }],
+                });
+                return done();
+            });
+        });
+    });
+
+    describe('_recursiveParseKafkaMessageValue', () => {
+        it('Should parse stringified object', done => {
+            const objStr = JSON.stringify(changeStreamDocument);
+            const parsed = listRecordStream._recursiveParseKafkaMessageValue(objStr);
+            assert.deepEqual(parsed, changeStreamDocument);
+            return done();
+        });
+        it('Should parse doubly stringified object', done => {
+            const objStr = JSON.stringify(JSON.stringify(changeStreamDocument));
+            const parsed = listRecordStream._recursiveParseKafkaMessageValue(objStr);
+            assert.deepEqual(parsed, changeStreamDocument);
+            return done();
+        });
+        it('Should return empty object when format is invalid', done => {
+            const parsed = listRecordStream._recursiveParseKafkaMessageValue('');
+            assert.deepEqual(parsed, {});
+            return done();
+        });
```

Comment:
None of these seem to need done.
```suggestion
        it('Should parse stringified object', () => {
            const objStr = JSON.stringify(changeStreamDocument);
            const parsed = listRecordStream._recursiveParseKafkaMessageValue(objStr);
            assert.deepEqual(parsed, changeStreamDocument);
        });
        it('Should parse doubly stringified object', () => {
            const objStr = JSON.stringify(JSON.stringify(changeStreamDocument));
            const parsed = listRecordStream._recursiveParseKafkaMessageValue(objStr);
            assert.deepEqual(parsed, changeStreamDocument);
        });
        it('Should return empty object when format is invalid', () => {
            const parsed = listRecordStream._recursiveParseKafkaMessageValue('');
            assert.deepEqual(parsed, {});
        });
```